lib/fluent/plugin/out_s3.rb

```diff
@@ -122,6 +123,8 @@ def desc(description)
     config_param :signature_version, :string, :default => nil # use nil to follow SDK default configuration
     desc "Given a threshold to treat events as delay, output warning logs if delayed events were put into s3"
     config_param :warn_for_delay, :time, :default => nil
+    desc "Custom temp dir for compressor, instead of system temp directory."
+    config_param :customer_tmp_dir, :string, :default => Dir.tmpdir
```

Comment:
Thanks for your reply. We used this plugin for two years, only got failure from a week ago. 

I checked my configuration, at 00:05(time_slice_wait 5m), lots of tag(about ~1G pre tag) use `/tmp` for chunk to file(~1G per tag) and compress(~0.4G pre tag), which will cost more than 10G space. It do not check whether if compress successful or failed, the compress like lzop will stop when have no space, and then upload the incomplete lzo file to s3, without error and retry.

and numbers of tag are keep growing, so I think it could be a good idea to mount a new disk as `tmp_dir`.

```
      buffer_type file
      buffer_path /mnt/fluentd_s3_buffer/__TAG__
      buffer_chunk_limit 1G

      time_slice_format %Y-%m-%d
      time_slice_wait 5m
```